src/Pomelo.EntityFrameworkCore.MySql/Extensions/MySqlDbContextOptionsExtensions.cs

```diff
@@ -87,5 +90,25 @@ private static MySqlOptionsExtension GetOrCreateExtension(DbContextOptionsBuilde
                 ? new MySqlOptionsExtension(existing)
                 : new MySqlOptionsExtension();
         }
+
+        private static bool IsAbleToUseDateTime6(string connectionStr)
```

Comment:
If we add a static class, we can remove `MySqlDesignTimeTypeMapper`. The static class should contains a property `IsCached`, if it is false, then call IsAbleToUseDateTime6. The type mapper maybe should add as Transient